providers/irdma/uverbs.c

```diff
@@ -206,19 +206,28 @@ int irdma_ubind_mw(struct ibv_qp *qp, struct ibv_mw *mw,
 		   struct ibv_mw_bind *mw_bind)
 {
 	struct ibv_mw_bind_info	*bind_info = &mw_bind->bind_info;
-	struct verbs_mr *vmr = verbs_get_mr(bind_info->mr);
-	struct irdma_umr *umr = container_of(vmr, struct irdma_umr, vmr);
+	struct verbs_mr *vmr;
+	struct irdma_umr *umr;
 
 	struct ibv_send_wr wr = {};
 	struct ibv_send_wr *bad_wr;
 	int err;
 
-	if (vmr->mr_type != IBV_MR_TYPE_MR || mw->type != IBV_MW_TYPE_1)
-		return ENOTSUP;
-
-	if (umr->acc_flags & IBV_ACCESS_ZERO_BASED)
+	if (!bind_info->mr && (bind_info->addr || bind_info->length))
 		return EINVAL;
 
+	if (bind_info->mr) {
```

Comment:
1. Why do we need these checks in user-space and not in the driver?
It is preferable to have one place for the checks and drivers are supposed to have these type of checks anyway.
2. I imagine almost all drivers need these checks. Why irdma is special here?
3. You have double SOB for Tatyana in the commit message.
Signed-off-by: Signed-off-by: Tatyana Nikolova <tatyana.e.nikolova@intel.com>

Reply:
> 1. Why do we need these checks in user-space and not in the driver?
>    It is preferable to have one place for the checks and drivers are supposed to have these type of checks anyway.
> 2. I imagine almost all drivers need these checks. Why irdma is special here?
> 3. You have double SOB for Tatyana in the commit message.
>    Signed-off-by: Signed-off-by: Tatyana Nikolova [tatyana.e.nikolova@intel.com](mailto:tatyana.e.nikolova@intel.com)

Hi Leon - Thanks for the review. Sindhu is out for the rest of the year. And will work on these review comments when she is back on Jan 4th.

Shiraz

Reply:
Thanks for the update.

Reply:
Bind isn't supposed to call to the kernel, but perhaps the checkes should be done in the core rdma-core code..

Reply:
Will send out a patch. Thanks Jason/Leon.